### ISSUE TITLE
Found a Bug with reshaping during self-attention

### ISSUE THREAD

**Issue by jefferythewind** [jwallbridge/translob#6]
In the LobAttention script, on line 112 `d_model` should be `d_submodel`, which I realized after comparing to the reference script here: https://github.com/kpot/keras-transformer/blob/master/keras_transformer/attention.py. Indeed the current code only functions correctly if the batch number happens to jibe correctly with the other dimensions: ie. 100 and 15, ie batch 32 doesn't work. Also, after 200 epoch in-sample accuracy maxed out around 78%, and val_accuracy much less. After fixing, I can run any batch number and code is working as expected :-)

**Comment by 772435284** [comment]
Can you reproduce the result in the paper after fixing the bug?

**Comment by jefferythewind** [comment]
I was able to get something to work resembling what was shown in the paper. However I found the results very sensitive to initialization. I did run a trial producing ~ 90% accuracy on the longest prediction target task.

**Comment by 772435284** [comment]
Cool, would you mind to give me a email address for me to communicate with you on the code? I am trying to make this code work and adapt it to pytorch.

**Comment by 772435284** [comment]
Can you share your parameters? I am unable to get 90% accuracy based on the paper

**Comment by VDCHHKK** [comment]
Hello Jeff, I have been trying to max out the accuracy in the TransLOB model but can't get past 60%, even after making your modification. Do you have a main.py model that I could compare mine to? 

Thank you very much for your time.

**Comment by LamprosGan** [comment]
@jefferythewind, did you really reproduce the results of the paper?

**Comment by jefferythewind** [comment]
@LamprosGan Yes I did, I've been meaning to share the results for some time. I followed along the paper using the building blocks in the repository and it was possible to get the same results posted in the paper. I will update here with the link once I post the code shortly.

**Comment by jefferythewind** [comment]
@LamprosGan @VDCHHKK @772435284 

Here you have it, the Translob code which replicates results in the paper. I will be adding more context around it. It was part of a recent paper which I will release on Arxiv about a data augmentation procedure which improves training time and performance on DeepLOB and (sometimes) TransLOB.

Please enjoy and let me know with any feedback/questions. Thanks.

https://github.com/jefferythewind/LOB-Data-Augmentation/blob/main/TransLOB.ipynb
